data_scrapers/utils.py

```diff
@@ -0,0 +1,25 @@
+from pathlib import Path
+import json
+from typing import Dict
+import requests
+from urllib.parse import urljoin
+
+def get_data_model() -> Dict:
+    """ Return a dictionary representation of the data model """
+    template_path = Path(__file__).parent.parent / 'data_models/data_model.json'
+    with template_path.open() as template:
+        out = json.load(template)
+    return out
+
+class SocrataApi:
+    """ Class for starting a Session with Socrata endpoints and sending requests"""
+    def __init__(self, base_url):
+        self.session = requests.Session()
+        self.base_url = base_url
+        self.resource_url = urljoin(self.base_url, '/resource/')
+        self.metadata_url = urljoin(self.base_url, '/api/views/metadata/v1/')
+
+    def request(self, url, **kwargs):
```

Comment:
It might be good to add some caching here for cases where the same URL is requested multiple times (e.g. at least two metadata functions grab the same set of URLs).

The most quick-n-dirty built-in you could use is `functools.lru_cache`, which will leak memory, but probably isn’t a huge deal for our current use cases:

```py
import functools

class SocrataApi:
    @functools.lru_cache()
    def request(self, url, **kwargs):
        ...
```

The [methodtools package](https://pypi.org/project/methodtools/) has a version of `lru_cache` that won’t leak.

A better long-term approach might be to use [cachecontrol](https://cachecontrol.readthedocs.io), though. It’s designed more precisely for caching with `requests`.